### ISSUE TITLE
platform_key for ARM

### ISSUE THREAD

**Issue by visr** [JuliaPackaging/BinaryProvider.jl#52]
Came across this in GEOSBuilder, building with `TARGET=arm-linux-gnueabihf`.
https://travis-ci.org/JuliaGeo/GEOSBuilder/jobs/360575777#L507-L516

```julia
julia> triplet(BinaryProvider.Linux(:armv7l))
"arm-linux-gnueabihf"

julia> platform_key("arm-linux-gnueabihf")
WARNING: Platform `arm-linux-gnueabihf` is not an officially supported platform
BinaryProvider.UnknownPlatform()

julia> platform_key("armv7l-linux-gnueabihf")
BinaryProvider.Linux(:armv7l, :glibc, :eabihf)
```

This can be reversed by chaning the [dict in platform_key](https://github.com/JuliaPackaging/BinaryProvider.jl/blob/e9dd1a8f39ba6ede973165512788cfa374ad7bf6/src/PlatformNames.jl#L283) to `:armv7l => "arm"`, but I'm not sure if this is the way to go, or that we need do deal with [sub-arch](https://github.com/rust-lang/rust/issues/33147#issuecomment-213635208) explicitly.

**Comment by staticfloat** [comment]
Yeah, my bad, I screwed this one up.  Fixed in #53

**Comment by visr** [comment]
That works, thanks!
